### ISSUE TITLE
`nix.optimise.automatic` and `nix.auto-optimise-store` do not work with `writeableStoreOverlay`

### ISSUE THREAD

**Issue by Sohalt** [astro/microvm.nix#50]
`nix.optimise.automatic` and `nix.auto-optimise-store` try to create hardlinks accross the layers of the overlayfs, which fails.

I think we should either have an `assert` with a helpful error message, or `lib.mkForce` these options to `false`, when using an overlay.

**Comment by astro** [comment]
Good idea. I'd prefer an assertion so that user configuration will not get overwritten quietly.
